src/test/java/com/guysfromusa/carsgame/v1/CarResourceTest.java

```diff
@@ -79,34 +84,53 @@ public void shouldRemoveCar(){
                 .until(() -> addNewCar(template, name, monsterType) != null);
 
         //when
-        ResponseEntity<Long> removedCarIdResponse = template.exchange(url, DELETE, null, Long.class);
+        Long id = removeCar(template, name);
 
         //then
-        assertThat(removedCarIdResponse.getBody()).isGreaterThan(0);
+        assertThat(id).isGreaterThan(0);
     }
 
     @Test
     public void shouldAddCarToGame(){
         //given
-        String name = "My-Sweet-Car";
+        String name = "car2";
+        String game = "game1";
+
+        String url = String.join("/", "/v1/cars", name, "cars", game);
+        Point point = new Point(1, 0);
+
+        //when
+        Car car = assignCarToTheGame(template, name, game, point);
+
+        //then
+        assertThat(car.getPosition())
+                .extracting(Point::getX, Point::getY)
+                .contains(1, 0);
+    }
+
+    @Test
+    public void shouldRejectAdditionCrashedCarToGame(){
+        //given
+        String name = "car3";
         String game = "game1";
-        String monsterType = "MONSTER";
 
-        String url = String.join("/", "/v1/cars", name, "game", game);
+        String url = String.join("/", "/v1/cars", name, "cars", game);
         Point point = new Point(1, 1);
 
         HttpEntity<Point> requestEntity = new RequestBuilder<Point>().body(point).build();
 
         //when
-        addNewCar(template, name, monsterType);
-        ResponseEntity<Car> carResponseEntity = template.exchange(url, POST, requestEntity, Car.class);
+        ResponseEntity<IllegalArgumentException> carResponse = template.exchange(url, POST, requestEntity, IllegalArgumentException.class);
```

Comment:
You should already here define that request should return ResponseEntity<ApiError> using exception handlers exceptions are transparent for client